examples/DGmethods/ex_001_dycoms.jl

```diff
@@ -0,0 +1,259 @@
+using MPI
+using CLIMA
+using CLIMA.Mesh.Topologies
+using CLIMA.Mesh.Grids
+using CLIMA.DGmethods
+using CLIMA.DGmethods.NumericalFluxes
+using CLIMA.MPIStateArrays
+using CLIMA.LowStorageRungeKuttaMethod
+using CLIMA.ODESolvers
+using CLIMA.GenericCallbacks
+using CLIMA.Atmos
+using CLIMA.VariableTemplates
+using CLIMA.MoistThermodynamics
+using CLIMA.PlanetParameters
+using LinearAlgebra
+using StaticArrays
+using Logging, Printf, Dates
+using CLIMA.Vtk
+
+@static if haspkg("CuArrays")
+  using CUDAdrv
+  using CUDAnative
+  using CuArrays
+  CuArrays.allowscalar(false)
+  const ArrayType = CuArray 
+else
+  const ArrayType = Array 
+end
+
+if !@isdefined integration_testing
+  const integration_testing =
+    parse(Bool, lowercase(get(ENV,"JULIA_CLIMA_INTEGRATION_TESTING","false")))
+end
+
+
+
+"""
+  Initial Condition for DYCOMS_RF01 LES
+"""
+function Initialise_DYCOMS!(state::Vars, aux::Vars, (x,y,z), t)
+    
+  DF         = eltype(state)
+  xvert::DF  = z
+
+  # Constants as defined in Stevens et. al (2005) 
+  R_d::DF     = 287.0
+  cp_d::DF    = 1015.0
+  cp_v::DF    = 1859.0
+  cp_l::DF    = 4181.0
+  Lv::DF      = 2.47e6
+  epsdv::DF   = 1.61
+  g::DF       = grav
+  p0::DF      = 1.0178e5
+  ρ0::DF      = 1.22
+  r_tot_sfc::DF=8.1e-3
+  Rm_sfc          = R_d * (1.0 + (epsdv - 1.0)*r_tot_sfc)
+  ρ_sfc::DF   = 1.22
+  P_sfc           = 1.0178e5
+  T_0::DF     = 285.0
```

Comment:
Maybe good to use `T_ref` or `T_BL` (for boundary layer) to avoid confusion with the reference temperature `T_0` that we use in internal energy definitions.